cmd/backfillEvents.go

```diff
@@ -0,0 +1,66 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/makerdao/vulcanizedb/libraries/shared/logs"
+	"github.com/makerdao/vulcanizedb/utils"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var endingBlockNumber int64
+
+// backfillEventsCmd represents the backfillEvents command
+var backfillEventsCmd = &cobra.Command{
+	Use:   "backfillEvents",
+	Short: "BackFill events from already-checked headers",
+	Long: `Fetch and persist events from configured transformers across a range
+of headers that may have already been checked for logs. Useful when adding a
+new event transformer to an instance that has already been running and marking
+headers checked as it queried for the previous (now incomplete) set of logs.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		err := backFillEvents()
+		if err != nil {
+			logrus.Fatalf("error back-filling events: %s", err.Error())
+		}
+		logrus.Info("completed back-filling events")
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(backfillEventsCmd)
+	backfillEventsCmd.Flags().Int64VarP(&endingBlockNumber, "ending-block-number", "e", -1, "last block from which to back-fill events")
+}
+
+func backFillEvents() error {
+	if endingBlockNumber == int64(-1) {
```

Comment:
You can mark the endingBlockNumber as required in the init() and you won't need this check - https://github.com/spf13/cobra/blob/2c5a0d300f8ba1cd142776508519aba999b5e77f/shell_completions.go#L10

Reply:
Nice 👍